mount.go

```diff
@@ -137,10 +138,49 @@ func doMount(args *argContainer) {
 	// Return memory that was allocated for scrypt (64M by default!) and other
 	// stuff that is no longer needed to the OS
 	debug.FreeOSMemory()
+	// Set up autounmount, if requested.
+	if args.autounmount > 0 && !args.reverse {
+		// Not being in reverse mode means we always have a forward file system.
+		fwdFs := fs.(*fusefrontend.FS)
+		go idleMonitor(args.autounmount, fwdFs, srv, args.mountpoint)
+	}
 	// Jump into server loop. Returns when it gets an umount request from the kernel.
 	srv.Serve()
 }
 
+// Based on the EncFS idle monitor:
+// https://github.com/vgough/encfs/blob/1974b417af189a41ffae4c6feb011d2a0498e437/encfs/main.cpp#L851
+const ActivityCheckIntervalSeconds = 10
```

Comment:
Can you use a fraction of idleTimeout instead? For automated tests, I would like to use sub-second timeouts (which is also why I want to use flagSet.DurationVar instead of minutes)

Reply:
Hmm...one potential issue there is that we could end up waiting up to idleTimeout / checkIntervalFraction extra if an access occurs just after we check for idleness. E.g., if the timeout is 10 hours and we check 5 times during the timeout period, an access right after we check for idle would make us consider the next 2 hours to be busy. Maybe idleTimeout / 4 with a cap at 2 minutes?